### ISSUE TITLE
BashApps that use `functools.partial` fail to run

### ISSUE THREAD

**Issue by WardLT** [Parsl/parsl#2233]
Creating a BashApp based on a function wrapped using `functools.partial` results in an `AttributeError: 'functools.partial' object has no attribute '__name__'` when executing the function.

**To Reproduce**
Run the script below

**Expected behavior**
No `AttributeError`

**Environment**
 - OS: ubuntu
 - Python version: 3.9
 - Parsl version: 1.2.0

**Distributed Environment**
- Where are you running the Parsl script from ? Laptop
- Where do you need the workers to run ? Same as Parsl script

**Example Code**

```python
from functools import partial
​
from parsl.config import Config
from parsl.executors import HighThroughputExecutor
from parsl.app.bash import BashApp
import parsl
​
​
config = Config(
    executors=[
        HighThroughputExecutor(label="htex", max_workers=2)
    ],
)
parsl.load(config)
​
​
def f(a: float, b: float) -> str:
    return f"echo {a} {b}"
​
​
g = partial(f, 1)
g.__name__ = 'test_fun'
app = BashApp(g)
​
print('func.__name__:', app.func.__name__)  # Proving the function has a name
print(app(2).result())
```

**Comment by WardLT** [comment]
This error is a probably a result from a known issue in the Dill serializer https://github.com/uqfoundation/dill/issues/275.

**Comment by WardLT** [comment]
Another note: serializing partial functions works with `PickleSerializer` but not `DillSerializer`

```python
g = partial(f, 1)
update_wrapper(g, f)

print('func.__name__:', g.__name__)
serializer = PickleSerializer()
new_g = serializer.deserialize(serializer.serialize(g))
print(new_g.__name__)  # Prints "f"

serializer = DillSerializer()
new_g = serializer.deserialize(serializer.serialize(g))
print(new_g.__name__)  # Crashes
```

**Comment by WardLT** [comment]
This is fixed by #2234. At least to the extent that most users will care
